### PR TITLE
Branch update model uml diagram

### LINKED ISSUE #201: Update Model UML diagram in DG

### PR DESCRIPTION
**Pull request by Green-Tea-123**
closes #201

### CODE REVIEW

File: docs/images/ModelClassDiagram.png
Comment (Quasant) [inline docs/images/ModelClassDiagram.png:1]: I havent touched the UML software yet so I'm not sure about the limitations, but the *filtered and *all around Job is very clustered. 

Sadly, seems like puml isn't drag and drop ahhahah..

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -115,18 +115,24 @@ How the parsing works:
...
+* stores the currently 'selected' `Person`, `Job`, and `Company` objects (e.g., results of a search query) as a 
+separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>`, 
+`ObservableList<Job>`, and `ObservableList<Company>` that can be 'observed' e.g. the UI can be bound to this list 
+so that the UI automatically updates when the data in the list change.
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
-* does not depend on any of the other three components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components)
+* does not depend on any of the other three components (as the `Model` represents data entities of the domain, 
+they should make sense on their own without depending on other components)
 
+[//]: # (Not sure what to do with the diagram and text below)
```
Comment (Quasant) [inline docs/DeveloperGuide.md:135]: I would say keep it for now, and maybe nearer to the end we can choose to delete it or not.  

We aren't using this alternative model anyway, and not sure if there is any benefit to switching to it.

**Review by Quasant — APPROVED** [review]
Overall LGTM.

File: docs/images/ModelClassDiagram.png
Comment (Green-Tea-123) [reply docs/images/ModelClassDiagram.png:1]: yeah sadly. This was about as ordered as I could've made it

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Green-Tea-123) [reply docs/DeveloperGuide.md:135]: Alright we should just look at the whole DG near the end of 1.5 and discuss diagrams like this.

**Review by wassilililily — APPROVED** [review]
LGTM

**Pull request merged**